src/test/java/com/eppo/sdk/helpers/RuleValidatorTest.java

```diff
@@ -126,6 +142,20 @@ void testMatchesAnyRuleWhenRuleMatches() {
         Assertions.assertEquals(rule, RuleValidator.findMatchingRule(subjectAttributes, rules).get());
     }
 
+    @DisplayName("findMatchingRule() when rule matches with semver")
+    @Test
+    void testMatchesAnyRuleWhenRuleMatchesWithSemVer() {
+        List<Rule> rules = new ArrayList<>();
+        Rule rule = createRule(new ArrayList<>());
+        addSemVerConditionToRule(rule);
+        rules.add(rule);
+
+        SubjectAttributes subjectAttributes = new SubjectAttributes();
+        subjectAttributes.put("version", EppoValue.valueOf("2.0.5"));
```

Comment:
Consider using semvers that require a true semver comparison (as opposed to a string comparison)
```java
condition1.value = EppoValue.valueOf("1.5.0");
```
```java
subjectAttributes.put("version", EppoValue.valueOf("1.15.5"));
```